proxy/src/telemetry/metrics/mod.rs

```diff
@@ -219,6 +212,15 @@ impl Metrics {
     }
 
     pub fn generate_report(&mut self) -> ReportRequest {
+        let latency_max_values: Vec<u32> =
```

Comment:
This is effectively static, right?  Should be able to not re-compute it for each report.